learning_resources/serializers.py

```diff
@@ -438,3 +468,90 @@ class Meta:
             "resource_readable_num",
             "resource_type",
         ]
+
+
+class UserListSerializer(serializers.ModelSerializer, WriteableTopicsMixin):
+    """
+    Simplified serializer for UserList model.
+    """
+
+    item_count = serializers.SerializerMethodField()
+
+    def get_item_count(self, instance) -> int:
+        """Return the number of items in the list"""
+        return getattr(instance, "item_count", None) or instance.resources.count()
+
+    def create(self, validated_data):
+        """Ensure that the list is created by the requesting user"""
+        request = self.context.get("request")
+        if request and hasattr(request, "user") and isinstance(request.user, User):
+            validated_data["author"] = request.user
+            topics_data = validated_data.pop("topics", [])
+            with transaction.atomic():
+                userlist = super().create(validated_data)
+                userlist.topics.set(
+                    models.LearningResourceTopic.objects.filter(id__in=topics_data)
+                )
+            return userlist
+        return None
+
+    def update(self, instance, validated_data):
```

Comment:
**Question:** Does the docstring here mean
1. "This method will check the list is authored by requesting user"
2. or does it mean "Be sure the list is authored by the requesting user before anything calls this method".
3. or maybe something else.

I don't see (1) happening... That's taken care of by the permission class, yes?

Reply:
It's making sure that if an API PATCH/POST request sends `{"author": <some_other_author_id>}", it will be overwritten by the id of the actual requestor.

```python
validated_data["author"] = request.user
```